Sources/SecureEnclave.swift

```diff
@@ -0,0 +1,181 @@
+                }
+            case .userCancelled:
+                if case .userCancelled = rhs {
+                    return true
+                } else {
+                    return false
+                }
+            case .itemNotFound:
+                if case .itemNotFound = rhs {
+                    return true
+                } else {
+                    return false
+                }
+            }
+        }
+    }
+    
+    // MARK: Internal Methods
+    
+    /// - parameter service: The service of the keychain slice we want to check if we can access.
+    /// - parameter identifier: A non-empty identifier that scopes the slice of keychain we want to access.
+    /// - returns: `true` if the keychain is accessible for reading and writing, `false` otherwise.
+    /// - note: Determined by writing a value to the keychain and then reading it back out.
+    internal static func canAccessKeychain(with service: Service, identifier: Identifier) -> Bool {
+        // To avoid prompting the user for Touch ID or passcode, create a Valet with our identifier and accessibility and ask it if it can access the keychain.
+        let noPromptValet: Valet
+        switch service {
+        case .standard:
+            noPromptValet = .valet(with: identifier, accessibility: .whenPasscodeSetThisDeviceOnly)
+        case .sharedAccessGroup:
+            noPromptValet = .sharedAccessGroupValet(with: identifier, accessibility: .whenPasscodeSetThisDeviceOnly)
+        }
+        
+        return noPromptValet.canAccessKeychain()
+    }
+    
+    /// - parameter object: A Data value to be inserted into the keychain.
+    /// - parameter key: A Key that can be used to retrieve the `object` from the keychain.
+    /// - parameter options: A base query used to scope the calls in the keychain.
+    /// - returns: `false` if the keychain is not accessible.
+    @discardableResult
+    internal static func set(object: Data, forKey key: String, options: [String : AnyHashable]) -> Bool {
+        // Remove the key before trying to set it. This will prevent us from calling SecItemUpdate on an item stored on the Secure Enclave, which would cause iOS to prompt the user for authentication.
+        _ = Keychain.removeObject(forKey: key, options: options)
+        
+        switch Keychain.set(object: object, forKey: key, options: options) {
+        case .success:
+            return true
+            
+        case .error:
+            return false
+        }
+    }
+    
+    /// - parameter key: A Key used to retrieve the desired object from the keychain.
+    /// - parameter userPrompt: The prompt displayed to the user in Apple's Face ID, Touch ID, or passcode entry UI.
+    /// - parameter options: A base query used to scope the calls in the keychain.
+    /// - returns: The data currently stored in the keychain for the provided key. Returns `nil` if no object exists in the keychain for the specified key, or if the keychain is inaccessible.
```

Comment:
"Returns `nil`..." -> "Returns `.itemNotFound` if no object exists in the keychain for the specified key, or `.userCancelled` if the user cancels the user-presence prompt."  (Same with `string(forKey:withPrompt:options:)` below)

Reply:
woof! this was in a bunch of spots.